src/linear_algebra/basis.lean

```diff
@@ -243,6 +243,35 @@ of_repr (f.symm.trans b.repr)
 
 end map
 
+section map_coeffs
+
+variables {R' : Type*} [semiring R'] [module R' M] (f : R ≃+* R') (h : ∀ c (x : M), f c • x = c • x)
```

Comment:
In your application, is `R'` naturally an algebra over `R`? If so, this could also be modeled with a scalar tower and the assumption that `algebra_map` is bijective.

Reply:
Not really, they are two different fields of fractions of the same integral domain. The scalar tower setup also seems sensible, maybe I'll try to get one out of the other.

Reply:
It seems we're too low down in the import hierarchy to talk about `algebra_map`s, so what I'll do is define a `basis.algebra_map_coeffs` that works in your suggested situation.